design/one-pager-stacks-versioning.md

```diff
@@ -0,0 +1,248 @@
+# Stack Versioning
+* Owner: Luke Weber (@lukeweber)
+* Reviewers: Crossplane Maintainers
+* Status: Draft, revision 1.0
+
+## Problem
+
+Stacks installer currently relies on installing a particular version of stack from a docker tag. The stack installer in
+ the future will be able to support stack update policy, maintenance windows, warnings on major updates, and more.
+ To build more complex features around stack version management we will need a common way to express versions that is
+ widely understood and standards based, so we will unify on best practices and
+ [semantic version 2.0](https://semver.org/spec/v2.0.0.html) as the Crossplane versioning standard.
+
+#### Didn't docker tags already solve this?
```

Comment:
Sorry if this was already discussed somewhere, but have we already considered a Docker registry-based implementation of the described release channel mechanism? I am imagining conventions and patterns which we would implement on top of the existing tagging functionality. For example:

* A registry's `/tags/list` [is very similar](https://docs.docker.com/registry/spec/api/#listing-image-tags) to the functionality of the `/all` url on a release channel
* Getting an individual tag from the registry (such as `stable`) is very similar to the `/latest` url on a release channel
* For historical lookups, the individual image shas which become the "head" of a release channel could be tagged with some convention for historical purposes, such as `beta-2019-10-10-0.0.1` and `beta-2019-10-17-0.0.2`